packages/contracts-bedrock/test/Preinstalls.t.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { CommonTest } from "test/setup/CommonTest.sol";
+import { Preinstalls } from "src/libraries/Preinstalls.sol";
+import { Bytes } from "src/libraries/Bytes.sol";
+import { console2 as console } from "forge-std/console2.sol";
+
+interface IEIP712 {
+    function DOMAIN_SEPARATOR() external view returns (bytes32);
+}
+
+/// @title PreinstallsTest
+contract PreinstallsTest is CommonTest {
+    /// @dev The domain separator commits to the chainid of the chain
+    function test_preinstall_permit2_domain_separator() external view {
+        bytes32 domainSeparator = IEIP712(Preinstalls.Permit2).DOMAIN_SEPARATOR();
+        bytes32 typeHash =
+            keccak256(abi.encodePacked("EIP712Domain(string name,uint256 chainId,address verifyingContract)"));
+        bytes32 nameHash = keccak256(abi.encodePacked("Permit2"));
+        uint256 chainId = block.chainid;
+        bytes memory encoded = abi.encode(typeHash, nameHash, chainId, Preinstalls.Permit2);
+        bytes32 expectedDomainSeparator = keccak256(encoded);
+        assertEq(domainSeparator, expectedDomainSeparator, "Domain separator mismatch");
+        assertEq(chainId, uint256(31337));
+        assertEq(domainSeparator, bytes32(0x4d553c58ae79a6c4ba64f0e690a5d1cd2deff8c6b91cf38300e0f2b76f9ee346));
+        // Warning the Permit2 domain separator as cached in the DeployPermit2.sol bytecode is incorrect.
```

Comment:
Can't we update it? hmm

Reply:
Bug in the upstream library test tooling that has existed forever. Don't think we need to change it. The mainnet L1 bytecode serves as a template, that passes tests when using chainID=1 itself, well enough.